lib/c-api/src/wasm_c_api/wasi/mod.rs

```diff
@@ -344,7 +345,98 @@ pub unsafe extern "C" fn wasi_get_wasi_version(module: &wasm_module_t) -> wasi_v
         .unwrap_or(wasi_version_t::INVALID_VERSION)
 }
 
-/// Takes ownership of `wasi_env_t`.
+/// Non-standard type wrapping `wasm_extern_t` with the addition of
+/// two `wasm_name_t` respectively for the module name and the name of
+/// the extern (very likely to be an import). This non-standard type
+/// is used by the non-standard `wasi_get_unordered_imports` function.
+#[allow(non_camel_case_types)]
+#[derive(Clone)]
+pub struct wasm_named_extern_t {
+    module: Box<wasm_name_t>,
+    name: Box<wasm_name_t>,
+    r#extern: Box<wasm_extern_t>,
```

Comment:
Note to self (and to you!): we should review the ownership here and how this type is used: this says all `wasm_name_t` structs are owned by us and `Cloning` this type will deep copy data that we own meaning the names and extern will need to be freed separately

Reply:
Yes, the type owns everything (meant by the usage of `Box` here). I don't see anything incorrect here, should I?

Reply:
Yeah, looks fine to me on further review!